src/v/raft/consensus.cc

```diff
@@ -529,20 +529,38 @@ ss::future<result<model::offset>> consensus::linearizable_barrier() {
     co_return ret_t(snapshot.linearizable_offset);
 }
 
+ss::future<result<replicate_result>> chain_stages(replicate_stages stages) {
+    return stages.request_enqueued.then_wrapped(
+      [f = std::move(stages.replicate_finished)](
+        ss::future<> enqueued) mutable {
+          if (enqueued.failed()) {
+              return f.discard_result()
+                .handle_exception([](const std::exception_ptr& e) {
```

Comment:
if `enqueued` failed do you need to consume `enqueued` so that it isn't the new abandoned exceptional future?